### PR TITLE
feat: added `DeleteById` method

### PR DESCRIPTION
**Pull request by mattia-fumo**

### CODE REVIEW

File: client.go [hunk 1]
```diff
@@ -181,3 +181,21 @@ func (c Client[Resource]) Create(ctx context.Context, resource Resource, options
...
+	if err != nil {
+		return err
+	}
+
+	if err := options.setOptionsInRequest(req); err != nil {
+		return err
+	}
+
+	resource := new(Resource)
+	if _, err := c.client.Do(req, resource); err != nil {
```
Comment (davidebianchi) [inline client.go:197]: The response of the resource is the resource itself?

**Review by davidebianchi — CHANGES_REQUESTED** [review]

File: client.go (hunk 1 above)
Comment (mattia-fumo) [reply client.go:197]: The response body is expected to be empty. So, it makes no difference where we store it. Maybe a `nil` would be more appropriate?

**Review by Danielecina — APPROVED** [review]

File: client.go (hunk 1 above)
Comment (davidebianchi) [reply client.go:197]: Yes, it should be a nil if we does not want to handle the response body

Comment (mattia-fumo) [reply client.go]: Done

**Pull request merged**